engine/src/main/java/org/destinationsol/game/screens/MainGameScreen.java

```diff
@@ -128,9 +130,16 @@
         buttonList.addElement(menuButton);
 
         mapButton = new UiTextButton().setDisplayName("Map")
-                                      .enableSound()
-                                      .setTriggerKey(gameOptions.getKeyMap())
-                                      .setOnReleaseAction(uiElement -> inputManager.changeScreen(solApplication.getGame().getScreens().mapScreen));
+                .enableSound()
+                .setTriggerKey(gameOptions.getKeyMap())
+                .setOnReleaseAction(uiElement -> {
+                    MapScreen screen = solApplication.getGame().getScreens().mapScreen;
+                    boolean isOn = inputManager.isScreenOn(screen);
+                    inputManager.changeScreen(solApplication.getGame().getScreens().mainGameScreen);
+                    if (!isOn) {
+                        inputManager.changeScreen(screen);
+                    }
+                });
 
```

Comment:
I would recommend removing this newline, to maintain consistency with other buttons.